lib/ts/recipe/emailpassword/api/implementation.ts

```diff
@@ -578,69 +580,122 @@ export default function getAPIImplementation(): APIInterface {
                   status: "WRONG_CREDENTIALS_ERROR";
               }
             | {
-                  status: "SIGN_IN_NOT_ALLOWED";
+                  status: "SIGN_IN_NOT_ALLOWED" | "SIGN_IN_FAILED";
                   reason: string;
               }
             | GeneralErrorResponse
         > {
             let email = formFields.filter((f) => f.id === "email")[0].value;
             let password = formFields.filter((f) => f.id === "password")[0].value;
 
-            let response = await options.recipeImplementation.signIn({ email, password, tenantId, userContext });
-            if (response.status === "WRONG_CREDENTIALS_ERROR") {
-                return response;
-            }
+            while (true) {
```

Comment:
do we need while(true) here? There deosn't seem to be an race condition issue..

Reply:
not required for signIn. removed.